proto/parca/query/v1alpha1/query.proto

```diff
@@ -257,6 +257,9 @@ message QueryRequest {
 
   // which runtime frames to filter out, often interpreter frames like python or ruby are not super useful by default
   optional RuntimeFilter runtime_filter = 10;
+
+  // which binary to filter by
+  optional string binary_filter = 11;
```

Comment:
this should be a repeated field, and I think we can future proof here a bit, and make this a repeated "filter" and a filter can then be of varying kinds, and the first one would be the binary filter, that way we can then define it to be multiple binaries (which in the Go code we would turn into a set and in the actual filter don't do a byte-match but instead check if the binary is in the set)